# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: editor/spellchecker/TextServicesDocument.h
# Commit: 77740303aae0
# Full Hash: 77740303aae0dcdb49e4bbc285810a68e9f05d9e
# Author: Masayuki Nakano <masayuki@d-toybox.com>
# Date: 2021-07-13 21:42:32
# Regressor Bug: 1718924
# File Overlap Count: 2
# Description:
#   Bug 1718924 - part 11: Create custom class of `nsTArray<UniquePtr<OffsetEntry>>` r=m_kato
#   
#   There are some methods in `TextServicesDocument` which work only with
#   `TextServicesDocument::mOffsetTable`.  Once we move such methods to custom
#   class of `nsTArray<UniquePtr<OffsetEntry>>`, we can make `TextServicesDocument`
# ==============================================================================

diff -r cce06fb41251 -r 77740303aae0 editor/spellchecker/TextServicesDocument.h
--- a/editor/spellchecker/TextServicesDocument.h	Tue Jul 13 10:02:12 2021 +0000
+++ b/editor/spellchecker/TextServicesDocument.h	Tue Jul 13 10:11:06 2021 +0000
@@ -52,13 +52,27 @@
     eNext,
   };
 
+  class OffsetEntryArray final : public nsTArray<UniquePtr<OffsetEntry>> {
+   public:
+    /**
+     * Init() initializes this array with aFilteredIter.
+     *
+     * @param[in]  aIterRange   Can be nullptr.
+     * @param[out] aAllTextInBlock
+     *                          Returns all text in the block.
+     */
+    Result<IteratorStatus, nsresult> Init(
+        FilteredContentIterator& aFilteredIter, IteratorStatus aIteratorStatus,
+        nsRange* aIterRange, nsAString* aAllTextInBlock = nullptr);
+  };
+
   RefPtr<dom::Document> mDocument;
   nsCOMPtr<nsISelectionController> mSelCon;
   RefPtr<EditorBase> mEditorBase;
   RefPtr<FilteredContentIterator> mFilteredIter;
   nsCOMPtr<nsIContent> mPrevTextBlock;
   nsCOMPtr<nsIContent> mNextTextBlock;
-  nsTArray<UniquePtr<OffsetEntry>> mOffsetTable;
+  OffsetEntryArray mOffsetTable;
   RefPtr<nsRange> mExtent;
 
   // TODO: Making the following members manged in a struct must become the code
@@ -295,11 +309,6 @@
   bool SelectionIsCollapsed() const;
   bool SelectionIsValid() const;
 
-  static nsresult CreateOffsetTable(
-      nsTArray<UniquePtr<OffsetEntry>>* aOffsetTable,
-      FilteredContentIterator* aFilteredIter, IteratorStatus* aIteratorStatus,
-      nsRange* aIterRange, nsAString* aStr);
-
   static nsresult NodeHasOffsetEntry(
       nsTArray<UniquePtr<OffsetEntry>>* aOffsetTable, nsINode* aNode,
       bool* aHasEntry, size_t* aEntryIndex);
